### PR TITLE
Use proper elixir project instead of compile time voodoo that isn't supposed to be used like it was

### PR DESCRIPTION
**Pull request by NobbZ**
This PR does the following chages:

1. Create another elixir project using `mix new`
2. Configure the project in a way that it _can_ produce a binary
3. Use that binary in run_commands
4. provide a proper binary-source-mapping
5. delete old elixir stuff and move the former described project at its place

These changes, so far, do not have brought any performance gains, but as a last addition, I incorporated changes which gave me another 10 to 15 seconds on my machine.

1. Wrap the line that outputs everything in a nasty and ugly looking call only to force direct usage of `:stdio` channel, instead of letting Elixir choose
2. Use `Regex.scan/3` instead of `String.split/3`

PS: I had to rebase twice before I got a green “Able to merge” on PR creation site, I do hope, that I haven't broken anything…

### CODE REVIEW

File: run_commands.txt [hunk 1]
```diff
@@ -20,3 +19,4 @@ python/wordcount_py2.py
 python/wordcount_py3.py
 rust/wordcount/wordcount
 scala -J-Xmx2g -classpath scala Wordcount
+elixir/wordcount
```
Comment (juditacs) [inline run_commands.txt]: Could you please sort it alphabetically?

**Comment by juditacs** [comment]
File changes look ok. I renamed/removed/reorganized a lot of stuff today, that was probably the reason for your problems.

File: run_commands.txt (hunk 1 above)
Comment (NobbZ) [inline run_commands.txt]: Of course I can

**Comment by juditacs** [comment]
There was indeed a noticable improvement:

| Rank | Experiment | CPU seconds | User time | Maximum memory | Contributor |
| :---: | :---: | :---: | :---: | :---: | :---: |
| 1 | elixir/wordcount | 397.52 | 387.1 | 2862204 | Norbert Melzer | 
| 2 | elixir elixir/wordcount.ex | 417.88 | 407.45 | 2545708 |  |

Anyways, I updated the second leaderboard. Elixir beats both javascript and typescript. The full test is running now, but I'm afraid it'll run out of memory.

**Comment by NobbZ** [comment]
May I ask what file you use for the tests or which specs your machine has?

I do fetch the hungarian wiki as described in the wiki using `scripts/create_input.sh`, and against this on my machine I had a user time of ~110 secs for the last benches, also peak-mem was at ~1.6GiB instead of your 2.8, so I'd guess you are using a different (bigger) datasource for this tests.

**Comment by juditacs** [comment]
Yes, the one you're using is Wikisource. I use Wikipedia for the full tests. There is a create_large_input.sh which downloads and unpacks it.

The small test is on the first 5M lines, the full test is on the full huwikipedia (not wikisource).

It's running on a VM in my university's cloud. 16GB, 4 cores, Ubuntu 14.04 LTS. The Docker runs Ubuntu 14.04 as well. I'm not really sure how to look up more.

**Comment by juditacs** [comment]
It did run out of memory.

**Comment by NobbZ** [comment]
I've another improvement in the pipeline. I already encountered a speed gain of roughly 90% against wikisource while memory was increased by about 300 MiB. Currently the fullest is running and has currently a memory consumption of roughly 8GiB and is in a state of constant swapping. It gets around 2 seconds of CPU time per wall clock minute. Current CPU time is at 25 minutes, I'll let it run for another 5 or 6 hours wall clock time or until my system enters an unusable state and report my observations in the PR that will follow.

**Comment by NobbZ** [comment]
I just killed the process.

`htop` told about 10.1 GiB of Virtual Mem used with ~7 GiB RES. The process was more or less stale for about an hour of wall time.

Using `eval cat ../data/huwiki-latest-pages-meta-current.xml | /usr/bin/time -f '%E_%U_%S__%M' ./wordcount > /dev/null` from within the `elixir` folder gave me the following output from time: `4:50:46_1495.45_83.09__7483068`

**Pull request merged**